### PR TITLE
[portconfig] Validate duplicate speed value and interface type value

### PR DESCRIPTION
**Pull request by Junchao-Mellanox**
#### What I did

Validate input parameter for command `config interface advertised-speeds` and `config interface advertised-types`, make sure there is no duplicate speeds or types

#### How I did it

Split input speeds/types with "," and check if there is a duplication, if yes, print an error.

#### How to verify it

1. new unit test case
2. manual test

#### New command output (if the output of a command-line utility has changed)
```
admin@sonice:$ sudo config interface advertised-speeds Ethernet4 100000,25000,100000
Invalid speed specified: 100000,25000,100000
Please remove duplicate speed
```

```
admin@sonic:$ sudo config interface advertised-types Ethernet4 CR,CR4,CR4
Invalid interface type specified: CR,CR4,CR4
Please remove duplicate interface type
```

### CODE REVIEW

**Review by keboliu — APPROVED** [review]

File: scripts/portconfig [hunk 1]
```diff
@@ -141,7 +141,12 @@ class portconfig(object):
             supported_speeds_str = self.get_supported_speeds(port)
             if supported_speeds_str:
                 supported_speeds = set(supported_speeds_str.split(','))
-                config_speeds = set(adv_speeds.split(','))
+                config_speed_list = [x.strip() for x in adv_speeds.split(',')]
+                config_speeds = set(config_speed_list)
+                if len(config_speeds) < len(config_speed_list):
+                    print('Invalid speed specified: {}'.format(adv_speeds))
```
Comment (liat-grozovik) [inline scripts/portconfig:147]: could we make it as one line of feedback to the user instead of 2?

File: scripts/portconfig [hunk 2]
```diff
@@ -164,7 +169,12 @@ class portconfig(object):
             print("Setting adv_interface_types %s on port %s" % (adv_interface_types, port))
         
         if adv_interface_types != 'all':
-            config_interface_types = set(adv_interface_types.split(','))
+            config_interface_type_list = [x.strip() for x in adv_interface_types.split(',')]
+            config_interface_types = set(config_interface_type_list)
+            if len(config_interface_types) < len(config_interface_type_list):
+                print("Invalid interface type specified: {}".format(adv_interface_types))
+                print('Please remove duplicate interface type')
```
Comment (liat-grozovik) [inline scripts/portconfig:176]: could we make it as one line of feedback to the user instead of 2?

**Review by liat-grozovik — APPROVED** [review]

**Comment by liat-grozovik** [comment]
@keboliu can you please signoff?

**Review by keboliu — APPROVED** [review]

**Review by qiluo-msft — APPROVED** [review]

**Pull request merged**